lib/dialects/sqlite/query-generator.js

```diff
@@ -79,8 +82,139 @@ const QueryGenerator = {
     return !!value ? 1 : 0;
   },
 
-  addColumnQuery(table, key, dataType) {
+  handleSequelizeMethod(smth, tableName, factory, options, prepend) {
+    if (smth instanceof Utils.Json) {
+      // Parse nested object
+      if (smth.conditions) {
+        const conditions = _.map(this.parseConditionObject(smth.conditions), condition =>
+          `json_extract(${this.quoteIdentifier(_.first(condition.path))}, '$.${_.tail(condition.path).join('.')}') = '${condition.value}'`
+        );
+
+        return conditions.join(' and ');
+      } else if (smth.path) {
+        let str;
+
+        // Allow specifying conditions using the sqlite json1 functions
+        // https://sqlite.org/json1.html
+        if (/(json(?:_.*)*)\(.*\)/i.test(smth.path)) {
```

Comment:
@sushantdhiman That seems to be a test limited to the GEOMETRY type.

Unfortunately, I really do not have any knowledge of what kind of injection is possible for JSON methods. Please let me know if there is any easy reference.